tests/MongoDB.Bson.Tests/IO/BsonBinaryReaderTests.cs

```diff
@@ -247,25 +247,30 @@ public void TestReadRawBsonDocument()
         [InlineData(GuidRepresentation.Unspecified)]
         public void ReadBinaryData_subtype_3_should_use_GuidRepresentation_from_settings(GuidRepresentation guidRepresentation)
         {
-            var settings = new BsonBinaryReaderSettings { GuidRepresentation = guidRepresentation };
-            var bytes = new byte[] { 29, 0, 0, 0, 5, 120, 0, 16, 0, 0, 0, 3, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 0 };
-            using (var stream = new MemoryStream(bytes))
-            using (var reader = new BsonBinaryReader(stream, settings))
+#pragma warning disable 618
+            if (BsonDefaults.GuidRepresentationMode == GuidRepresentationMode.V2)
```

Comment:
A few tests have been changed to only run in V2 mode (like this test).

But most have been tweaked to pass in either V3 mode or V2 mode with any Guid representation.